### PR TITLE
Coverage, hardening, and homepage CLS fix: controller tests and refactor, parallel dashboard fetches, param whitelists, docs cleanup, condition-alerts PRD

### PR DESCRIPTION
**Pull request by jelrod27**
## Summary

Batch of 11 reviewed changes produced by a full-codebase audit, executed as a linear chain of 13 commits. Each change was implemented in an isolated worktree and independently verified before stacking.

### Tests and coverage (511 -> 579 tests)

- useWeatherController characterization suite (14 tests): rate limiting, unit-scoped search cache keys, stale-response loadId guards
- Open-Meteo adapter hourly-window edge cases: truncated arrays, missing utc_offset_seconds, all-past fallback, empty input
- /api/stargazer route-level tests: param validation, upstream-failure codes, 7Timer/ISS/geocode degradation, timezone-suffix sign handling
- getProfile/updateProfile error-fallback branching (pins that updateProfile's column-missing fallback is a same-query retry)
- formatHeaderLocation extracted from navigation.tsx to lib/ and branch-tested; Navigation mobile-menu smoke tests
- LocationCard tests including a fetch-concurrency pin

### Behavior changes

- Dashboard location card: four sequential detailed-weather fetches replaced with one Promise.allSettled batch (parallel; identical URLs, error strings, and degradation semantics)
- API input hardening: units query param whitelisted in weather/current and weather/forecast (closes the upstream parameter-injection class flagged in the 2026-05 security sweep); RSS categories validated against the FeedCategory union
- Homepage CLS fix: welcome/skeleton swap made atomic and RandomCityLinks deferred until the auto-location flow settles. Lighthouse CLS median 0.33 -> 0.022, perf median 0.99 (5-run lhci against a local production build with real env). The live site currently measures CLS 0.269 from the same root cause, so this should lift deployed perf off the 0.85 gate as well.

### Refactor

- useWeatherController: 552 -> 416 lines, 13 -> 3 useCallbacks. Client rate limiter and search cache extracted to lib/weather-rate-limit.ts and lib/weather-search-cache.ts. Behavior pinned by the characterization suite before and after; localStorage keys and the hook's return shape unchanged (one additive field: autoLocationAttempted).

### Docs and repo truth

- Removed stale AI-assistant claims from package.json description and README; corrected CLAUDE.md git-hooks section (hooks are gitleaks secret scans, not E2E + Lighthouse); archived the orphaned AI UX plan; deleted three AI-only .skip migrations; added an npm run typecheck script
- New PRD: planning/prds/PRD-condition-alerts.md — design spike for location-scoped alerts (in-app channel recommended for v1; news-geography matching ruled out for v1 because feed items carry free-text locations only)

## Verification

- Jest (test:ci): 579/579
- tsc --noEmit: clean; ESLint: 0 errors
- Playwright full suite (Chromium): 42 passed, 1 skipped
- Production build: clean
- Lighthouse (5 runs, local prod server, real env): perf median 0.99, CLS median 0.022 — main measures 0.81-0.83 perf / CLS ~0.33 on the same machine
- gitleaks across all commits: no leaks

## Notes for reviewers

- CI's Lighthouse job builds with placeholder API keys, so it renders the no-data homepage and cannot regression-test the CLS fix; the numbers above were measured with a real env. A mock-weather mode for that job would close the blind spot.
- Residual CLS outlier (~1 in 5 runs, ~0.24): WeatherDisplay's dynamic-import h-96 placeholder expanding under slow chunk timing — pre-existing and separate; candidate follow-up.
- updateProfile's column-missing fallback re-runs an identical query; now pinned by tests, simplification deliberately deferred.

Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Concurrent weather data fetching for faster detailed weather loads
  * Enhanced auto-location detection flow with improved visual experience

* **Improvements**
  * Better location formatting in header display
  * Graceful handling of optional weather data failures (UV index and air quality fall back to default values)

* **Removed**
  * AI assistant feature

### CODE REVIEW

File: lib/api/query-params.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * an empty array when the param is present but contains no valid category
+ * (preserving the current "no matching feeds" behavior for garbage input).
+ */
+export function parseFeedCategories(raw: string | null): FeedCategory[] | undefined {
+  if (!raw) return undefined
+  return raw
+    .split(',')
+    .map((c) => c.trim())
+    .filter((c): c is FeedCategory => VALID_CATEGORIES.has(c))
+}
```
[the remark this answers is not in the document]
Comment (jelrod27) [reply lib/api/query-params.ts:36]: Deliberately not applied: `?categories=` → `undefined` → "all feeds" is the pre-existing mainline behavior (the old inline parse's `if (categoriesParam)` guard did exactly this). Changing `''` to return `[]` would flip an empty param from "all feeds" to "zero feeds" — a behavior regression. The docstring was the actual defect and is corrected in ed4986a; the empty-string behavior is now pinned by a test.

File: __tests__/query-params.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  })
+
+  it('returns empty array for entirely bogus input', () => {
+    expect(parseFeedCategories('bogus,<script>')).toEqual([])
+  })
+
+  it('filters out unknown values while keeping valid ones', () => {
+    expect(parseFeedCategories('severe,bogus')).toEqual(['severe'])
+  })
+})
```
[the remark this answers is not in the document]
Comment (jelrod27) [reply __tests__/query-params.test.ts:53]: Test added in ed4986a — it pins the pre-existing behavior (`''` → `undefined` → all feeds) rather than the suggested `[]`; see the thread on lib/api/query-params.ts for the rationale.

File: __tests__/weather-search-cache.test.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            {
+                day: 'Mon',
+                highTemp: 75,
+                lowTemp: 60,
+                condition: 'clear',
+                description: 'Clear sky',
+            },
+        ],
+        ...overrides,
+    } as WeatherData
```
[the remark this answers is not in the document]
Comment (jelrod27) [reply __tests__/weather-search-cache.test.ts:42]: Skipping intentionally: the cache module is payload-opaque — it JSON-roundtrips the stored object and strips only `coordinates`; it never reads `.city`/`.location` or any weather field. The fixture's field names don't change what these tests verify (roundtrip, expiry, unit-scoped keys, coordinate stripping), so aligning it with the full WeatherData shape would be cosmetic.

File: planning/prds/PRD-condition-alerts.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+- Geocoding the free-text location string (adds latency and a geocoding service), or
+- Parsing NWS CAP/Atom FIPS zone codes (available in the raw XML but not currently extracted by the parser).
+
+**Conclusion: news alerts are out of scope for v1.** This is the expected outcome; no STOP condition triggered.
+
+### 4.4 Alert-fatigue control
+
+- **Score threshold:** Default 70 ("Good" label per `getScoreLabel` in `lib/stargazer/score.ts`). User-configurable to 50 ("Fair") or 85 ("Excellent") in subscription settings.
+- **Dedup window:** No more than one `stargazing_window` alert per `(user_id, saved_location_id)` pair within a 7-day rolling window. The cron route checks `user_alerts` for a recent alert before writing a new one.
+- **Evaluation window:** Cron runs once daily at 14:00 UTC. The "upcoming night" is defined as the next dark window calculated from `lib/stargazer/astronomy.ts`'s `calculateDarkWindow`.
```
[the remark this answers is not in the document]
Comment (jelrod27) [reply planning/prds/PRD-condition-alerts.md:176]: Addressed in ed4986a — §4.4 now explicitly states a bare pre-insert SELECT is not race-safe against overlapping cron runs and requires an atomic guard (unique partial index on (user_id, saved_location_id) scoped to the active window, or pg_advisory_xact_lock).
